src/H5TSint.c

```diff
@@ -174,21 +244,45 @@ H5TS_term_package(void)
 herr_t
 H5TS__api_mutex_acquire(unsigned lock_count, bool *acquired)
 {
+#ifdef H5_HAVE_CONCURRENCY
+    unsigned dlftt = 0;
+#endif
     herr_t ret_value = SUCCEED;
 
     FUNC_ENTER_PACKAGE_NAMECHECK_ONLY
 
+#ifdef H5_HAVE_THREADSAFE
     /* Attempt to acquire the lock */
     if (H5_UNLIKELY(H5TS_mutex_trylock(&H5TS_api_info_p.api_mutex, acquired) < 0))
         HGOTO_DONE(FAIL);
 
-    /* If acquired, increment the levels of recursion by 'lock_count' - 1 */
+    /* If acquired, acquire the mutex ('lock_count' - 1) more times */
     if (*acquired) {
         for (unsigned u = 0; u < (lock_count - 1); u++)
             if (H5_UNLIKELY(H5TS_mutex_lock(&H5TS_api_info_p.api_mutex) < 0))
                 HGOTO_DONE(FAIL);
         H5TS_api_info_p.lock_count += lock_count;
     }
+#else /* H5_HAVE_CONCURRENCY */
```

Comment:
Implement the DLFTT protocol - don't reacquire global lock when locking is disabled for a thread.